### ISSUE TITLE
testing on train images but no detections

### ISSUE THREAD

**Issue by riyaj8888** [eriklindernoren/PyTorch-YOLOv3#168]
these are results during .. i tested with saved checkpoints on trained images itself but no detections and few detections are negative and some are larger than image size itself .. what does that mean ? why it happened ? help appreciated 
---- [Epoch 511/1200, Batch 147/293] ----
+------------+--------------+--------------+--------------+
| Metrics    | YOLO Layer 0 | YOLO Layer 1 | YOLO Layer 2 |
+------------+--------------+--------------+--------------+
| grid_size  | 13           | 26           | 52           |
| loss       | 0.161417     | 0.075214     | 0.044303     |
| x          | 0.000230     | 0.000552     | 0.002527     |
| y          | 0.000342     | 0.000666     | 0.001026     |
| w          | 0.020693     | 0.019103     | 0.020169     |
| h          | 0.002797     | 0.001406     | 0.001462     |
| conf       | 0.137354     | 0.053487     | 0.019118     |
| cls        | 0.000000     | 0.000000     | 0.000000     |
| cls_acc    | 100.00%      | 100.00%      | 100.00%      |
| recall50   | 1.000000     | 1.000000     | 1.000000     |
| recall75   | 1.000000     | 1.000000     | 1.000000     |
| precision  | 1.000000     | 0.875000     | 0.875000     |
| conf_obj   | 0.932152     | 0.981926     | 0.996375     |
| conf_noobj | 0.000364     | 0.000185     | 0.000068     |
+------------+--------------+--------------+--------------+
Total loss 0.2809331715106964
---- ETA 0:00:49.672057

**Comment by nswei** [comment]
Have you solve the problem?I also get some boxes much larger than the ground truth boxes

**Comment by Andy3525** [comment]
I get the same problem, too. Have you solve it ? thannnnnnks

**Comment by pieterbl86** [comment]
> I get the same problem, too. Have you solve it ? thannnnnnks

Did you try this code:
python3 detect.py --image_folder data/valid --checkpoint_model checkpoints/y
olov3_ckpt_99.pth --model_def config/yolov3-custom.cfg --class_path data/custom/classes.names --conf_thres 0.95 --nms_thres 0.4 **--weights_path checkpoints/yolov3_ckpt_99.pth**

If the weights path is not defined in your code then it will use the default (yolov3.weights). Refer to line 46 of detect.py 

`if opt.weights_path.endswith(".weights"):`
`# Load darknet weights`
`model.load_darknet_weights(opt.weights_path)`
`else:`
`# Load checkpoint weights`
`model.load_state_dict(torch.load(opt.weights_path))`

**Comment by Flova** [comment]
I closed this issue due to inactivity. Feel free to reopen for further discussion.
